### PR TITLE
Pro 4962 required if field

### PR DESCRIPTION
**Pull request by falkodev**
# As a developer, I want to be able to make a required field optional based on a user's selection in another field.

https://linear.app/apostrophecms/issue/PRO-4962/as-a-developer-i-want-to-be-able-to-make-a-required-field-optional

## Syntax 

- exact match on a value (boolean, string, number)
- logical AND by default
- logical OR
- not equal with exact match
- all the above with nested properties

```js
requiredIf: {
  fieldA: true,
  fieldB: 'active',
  $or: [
    { fieldC: 10 },
    { fieldD: 20 }
  ],
  fieldF: { $ne: 'default' },
  'fieldE.subFieldA': 'AAA'
}
```

## Acceptance criteria

- we have conditionally required fields on the backend
- it accepts the same syntax as if
- integration test

### CODE REVIEW

File: modules/@apostrophecms/schema/index.js [hunk 1]
```diff
@@ -465,6 +465,57 @@ module.exports = {
...
+              continue;
+            } else if (val.$ne) {
+              // eslint-disable-next-line eqeqeq
+              if (val.$ne == destinationKey) {
+                result = false;
+                break;
+              }
+            }
+
+            if (val.min) {
```
Comment (ValJed) [inline modules/@apostrophecms/schema/index.js:494]: If we support it for `requiredIf`, you should add it to `if` too?

**Review by ValJed — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/schema/index.js (hunk 1 above)
Comment (boutell) [reply modules/@apostrophecms/schema/index.js:494]: All features should work equally for `if` and `requiredIf`.

Comment (boutell) [reply modules/@apostrophecms/schema/index.js:494]: With as little code duplication as possible front and back end.

Comment (falkodev) [reply modules/@apostrophecms/schema/index.js:494]: done

**Review by ValJed — APPROVED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**